cirq-core/cirq/sim/clifford/clifford_simulator.py

```diff
@@ -132,8 +129,11 @@ def __init__(
         )
 
     @property
-    def final_state(self):
-        return self._final_simulator_state
+    def final_state(self) -> 'cirq.CliffordState':
```

Comment:
I'm not entirely clear on how this compares to the previous behavior. It looks like the new implementation generates a new `CliffordState` from a copy of the final state, whereas the old version just returned the final state directly. Is this what we want?

Reply:
The only functional difference is that this creates a new `CliffordState` object with each call. If we cache that here, (`@lru_cache`?) then the behavior is identical.

Reply:
I'm going to leave it as is, since CliffordStepResult is implemented the same way. lru_cache apparently isn't for instance methods, and it seems overkill to cache this manually as there's no reason to call it more than once. Feel free to reopen if you'd prefer to cache this.